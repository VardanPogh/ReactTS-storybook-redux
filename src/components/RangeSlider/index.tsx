import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import Slider from '@material-ui/core/Slider'

const useStyles = makeStyles({
  slider: {
    color: '#014bad',
  },
})

interface Props {
  value: number | number[]
  onChange(event: any, newValue: number | number[]): void
  getAriaValueText?(): string
}

const RangeSlider: React.FC<Props> = ({
  value,
  onChange,
  getAriaValueText,
}) => {
  const classes = useStyles()

  return (
    <Slider
      className={classes.slider}
      value={value}
      onChange={onChange}
      valueLabelDisplay="on"
      aria-labelledby="range-slider"
      getAriaValueText={getAriaValueText}
    />
  )
}

export default RangeSlider
