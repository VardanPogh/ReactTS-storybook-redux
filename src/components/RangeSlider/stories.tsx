import React from 'react'

import RangeSlider from '.'

export default {
  title: 'RangeSlider',
  component: RangeSlider,
}

export const Default: React.FC = () => {
  const [value, setValue] = React.useState<number[]>([20, 37])
  const handleChange = (event: any, newValue: number | number[]): void => {
    setValue(newValue as number[])
  }
  return (
    <div style={{ padding: 20 }}>
      <p style={{ marginBottom: 40 }}>Sample Range Slider</p>
      <RangeSlider value={value} onChange={handleChange} />
    </div>
  )
}
