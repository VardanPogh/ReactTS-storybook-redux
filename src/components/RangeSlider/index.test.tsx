import React from 'react'
import { render } from '@testing-library/react'

import RangeSlider from '.'

describe('RangeSlider Test', () => {
  test('Load RangeSlider', () => {
    const value = [20, 37]
    const myMock = jest.fn()
    const { container } = render(
      <RangeSlider value={value} onChange={myMock} />
    )
    expect(container).toMatchSnapshot()
  })
})
