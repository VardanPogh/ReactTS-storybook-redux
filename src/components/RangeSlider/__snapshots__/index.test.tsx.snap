// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeSlider Test Load RangeSlider 1`] = `
<div>
  <span
    class="MuiSlider-root MuiSlider-colorPrimary makeStyles-slider-1"
  >
    <span
      class="MuiSlider-rail"
    />
    <span
      class="MuiSlider-track"
      style="left: 20%; width: 17%;"
    />
    <input
      type="hidden"
      value="20,37"
    />
    <span
      aria-labelledby="range-slider"
      aria-orientation="horizontal"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="20"
      class="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-open-3 PrivateValueLabel-thumb-2"
      data-index="0"
      role="slider"
      style="left: 20%;"
      tabindex="0"
    >
      <span
        class="PrivateValueLabel-offset-4 MuiSlider-valueLabel"
      >
        <span
          class="PrivateValueLabel-circle-5"
        >
          <span
            class="PrivateValueLabel-label-6"
          >
            20
          </span>
        </span>
      </span>
    </span>
    <span
      aria-labelledby="range-slider"
      aria-orientation="horizontal"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="37"
      class="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-open-3 PrivateValueLabel-thumb-2"
      data-index="1"
      role="slider"
      style="left: 37%;"
      tabindex="0"
    >
      <span
        class="PrivateValueLabel-offset-4 MuiSlider-valueLabel"
      >
        <span
          class="PrivateValueLabel-circle-5"
        >
          <span
            class="PrivateValueLabel-label-6"
          >
            37
          </span>
        </span>
      </span>
    </span>
  </span>
</div>
`;
